### DISCUSSION TITLE
HeatSource and DerivativeParsedMaterial

### DISCUSSION THREAD

**Discussion by hamedattariani** [idaholab/moose#17402] [Q&A Modules: General] [answered]
We are solving two coupled transient heat conduction differential equations. The heat source terms (body force) in each equation are functions of temperatures, e.g. A*(T1-T2). So, we used "HeatSource" along with "DerivativeParsedMaterial" to define the source terms. This implementation does not work, because we see temperature rise, although one sets the heat source to zero. I think this implementation messes up the weak forms. Do I need to write a new kernel for this body force or there is already a kernel for this purpose in MOOSE?
 
[./volumetric_heat] 
    type = DerivativeParsedMaterial
    f_name = 'volumetric_heat1'
    args = 'T1 T2'
    function ='(1000)*(T1-T2)'
[../]

**Comment by GiudGiud** [comment, marked as answer]
Hello

there should be an ‘ambient convection’ kernel of this form in the navier stokes module you may use. I can’t now but I ll have a look later to see if I can post a link to it here

Guillaume

**Reply by hamedattariani** [reply]
I just looked at the Kernel. "T_ambient" is defined as a constant. In our case, T_ambient is the main variable itself.
The system of equations is as follow:

Heat capacity*d(T1/dt)=...**+(T1-T2)**
Heat capacity*d(T2/dt)=...**-(T1-T2)**

**Reply by GiudGiud** [reply]
ok then you could implement a version with non-constant T_ambient but a coupledVar instead. 

The finite volume implementation, NSFVEnergyAmbientConvection has it set up that way
